### PR TITLE
DDCE-3538: Upgrade Scala to use 2.13

### PR DESCRIPTION
**Pull request by rreddy80**
# PAS-DDCE-3538

##Scala upgrade

Scala version has been updated to 2.13.8
Silencer plugin has been removed
withSCoverage() is not using scala version param
All automated tests pass

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date
 - [ ]  I've added the links for relevant PRs for AT/PT in description

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date
 - [ ]  I've verified the links for relevant PRs for AT/PT in description before approval

### CODE REVIEW

File: app/repositories/LockRepository.scala [hunk 1]
```diff
@@ -21,7 +21,6 @@ import models.Lock
 import org.mongodb.scala.model.Filters.equal
 import org.mongodb.scala.model.{IndexModel, IndexOptions}
 import org.mongodb.scala.model.Indexes.{ascending}
```
Comment (lmugnai) [inline app/repositories/LockRepository.scala:23]: no need for braces here. Please remove.

File: run_all_tests.sh [hunk 2]
```diff
@@ -1,2 +1,2 @@
 #!/usr/bin/env bash
-sbt clean compile scalafmtAll scalastyleAll coverage test it:test dependencyUpdates coverageReport
\ No newline at end of file
+sbt clean compile scalafmtAll scalastyleAll coverage Test/test IntegrationTest/test dependencyUpdates coverageReport
```
Comment (lmugnai) [inline run_all_tests.sh:2]: add new line ad end of file

File: app/controllers/DeclarationController.scala [hunk 3]
```diff
@@ -92,7 +92,7 @@ class DeclarationController @Inject() (
                 case State.SubmissionFailed =>
                   Future.successful(Conflict)
                 case _                      =>
-                  paymentNotification.status match {
+                  (paymentNotification.status: String @unchecked) match {
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply app/controllers/DeclarationController.scala:95]: This is to stop unchecked compiler warnings

File: app/controllers/DeclarationController.scala [hunk 4]
```diff
@@ -137,7 +137,7 @@ class DeclarationController @Inject() (
         case State.SubmissionFailed =>
           Future.successful(Conflict)
         case _                      =>
-          paymentNotification.status match {
+          (paymentNotification.status: String @unchecked) match {
```
[the remark this answers is not in the document]
Comment (lmugnai) [reply app/controllers/DeclarationController.scala:140]: This is to stop unchecked compiler warnings

File: app/repositories/LockRepository.scala (hunk 1 above)
Comment (rreddy80) [reply app/repositories/LockRepository.scala]: fixed it now 👍

File: run_all_tests.sh (hunk 2 above)
Comment (rreddy80) [reply run_all_tests.sh]: fixed it now 👍

**Review by lmugnai — APPROVED** [review]

**Pull request merged**